aea/runtime.py

```diff
@@ -156,28 +180,24 @@ def _start(self) -> None:
         """
         self.set_loop(self._loop)
 
-        self._state.set(RuntimeStates.started)
-
         logger.debug(f"Start runtime event loop {self._loop}: {id(self._loop)}")
         self._task = self._loop.create_task(self.run_runtime())
 
         try:
             self._loop.run_until_complete(self._task)
-            self._state.set(RuntimeStates.loop_stopped)
             logger.debug("Runtime loop stopped!")
         except Exception:
             logger.exception("Exception raised during runtime processing")
+            self._state.set(RuntimeStates.error)
             raise
         finally:
             self._stopping_task = None
 
     async def run_runtime(self) -> None:
         """Run agent and starts multiplexer."""
+        self._state.set(RuntimeStates.starting)
         try:
-            self._state.set(RuntimeStates.starting)
-            self._agent.setup_multiplexer()
             await self._start_multiplexer()
-            self._agent.start_setup()
```

Comment:
@solarw and @marcofavorito ; I would have thought we should have `starting` where we call setup on multiplexer and agent, then when finished setting up, call `start` on both of them.

Reply:
isn't this the case already? Line 198 sets the state to `starting`